lib/cea/cea608_memory.js

```diff
@@ -0,0 +1,513 @@
+      if (this.rows_[i].length) {
+        lastNonEmptyRow = i;
+        break;
+      }
+    }
+
+    // Exit early if no non-empty row was found.
+    if (firstNonEmptyRow === -1 || lastNonEmptyRow === -1) {
+      return;
+    }
+
+    // Keeps track of the current styles for a cue being emitted.
+    let currentUnderline = false;
+    let currentItalics = false;
+    let currentTextColor = shaka.cea.Cea608Memory.DEFAULT_TXT_COLOR;
+    let currentBackgroundColor = shaka.cea.Cea608Memory.DEFAULT_BG_COLOR;
+
+    // Create first cue that will be nested in top level cue. Default styles.
+    let currentCue = this.createStyledCue_(startTime, endTime, '',
+        currentUnderline, currentItalics,
+        currentTextColor, currentBackgroundColor);
+
+    // Logic: Reduce rows into a single top level cue containing nested cues.
+    // Each nested cue corresponds to a style change. \n is used for new lines.
+    const topLevelCue = new shaka.text.Cue(startTime, endTime, '');
+
+    for (let i = firstNonEmptyRow; i <= lastNonEmptyRow; i++) {
+      for (const styledChar of this.rows_[i]) {
+        const underline = styledChar.isUnderlined();
+        const italics = styledChar.isItalicized();
+        const textColor = styledChar.getTextColor();
+        const backgroundColor = styledChar.getBackgroundColor();
+
+        // If any style properties have changed, we need to open a new cue.
+        if (underline != currentUnderline || italics != currentItalics ||
+            textColor != currentTextColor ||
+            backgroundColor != currentBackgroundColor) {
+          // Push the currently built cue and start a new cue, with new styles.
+          if (currentCue.payload) {
+            topLevelCue.nestedCues.push(currentCue);
+          }
+          currentCue = this.createStyledCue_(startTime, endTime, '',
+              underline, italics, textColor, backgroundColor);
+
+          currentUnderline = underline;
+          currentItalics = italics;
+          currentTextColor = textColor;
+          currentBackgroundColor = backgroundColor;
+        }
+
+        currentCue.payload += styledChar.getChar();
+      }
+      if (currentCue.payload) {
+        topLevelCue.nestedCues.push(currentCue);
+      }
+
+      // Create and push a linebreak cue to create a new line.
+      if (i !== lastNonEmptyRow) {
+        const spacerCue = new shaka.text.Cue(startTime, endTime, '');
+        spacerCue.spacer = true;
+        topLevelCue.nestedCues.push(spacerCue);
+      }
+
+      // Create a new cue.
+      currentCue = this.createStyledCue_(startTime, endTime, '',
+          currentUnderline, currentItalics,
+          currentTextColor, currentBackgroundColor);
+    }
+
+    if (topLevelCue.nestedCues.length) {
+      this.decoder_.addParsedClosedCaption({
+        cue: topLevelCue,
+        stream,
+      });
+    }
+  }
+
+  /**
+   * @param {!number} startTime
+   * @param {!number} endTime
+   * @param {!string} text
```

Comment:
All callers seem to pass a blank string here, so I would suggest simply removing this parameter.